sdstore/sdstore_test.go

```diff
@@ -670,13 +670,13 @@ func TestCheckForRetry(t *testing.T) {
 		{statusCode: http.StatusAccepted, doesRetry: false},
 		{statusCode: http.StatusNoContent, doesRetry: false},
 		// status 400~
-		{statusCode: http.StatusBadRequest, doesRetry: true},
-		{statusCode: http.StatusUnauthorized, doesRetry: true},
-		{statusCode: http.StatusForbidden, doesRetry: true},
+		{statusCode: http.StatusBadRequest, doesRetry: true, retryErr: fmt.Errorf("got code 400.")},
```

Comment:
except for 429 there is no need to retry for any 4xx errors

Reply:
makes sense! maybe also allow `408`? 

from wiki: `408 Request Timeout
The server timed out waiting for the request. According to HTTP specifications: "The client did not produce a request within the time that the server was prepared to wait. The client MAY repeat the request without modifications at any later time`

Reply:
addressed!